exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/metric_exporter/__init__.py

```diff
@@ -211,33 +235,468 @@ def export(
         if self._shutdown:
             _logger.warning("Exporter already shutdown, ignoring batch")
             return MetricExportResult.FAILURE
-        serialized_data = encode_metrics(metrics_data).SerializeToString()
+
+        serialized_data = encode_metrics(metrics_data)
         deadline_sec = time() + self._timeout
-        for retry_num in range(_MAX_RETRYS):
-            resp = self._export(serialized_data, deadline_sec - time())
-            if resp.ok:
-                return MetricExportResult.SUCCESS
-            # multiplying by a random number between .8 and 1.2 introduces a +/20% jitter to each backoff.
-            backoff_seconds = 2**retry_num * random.uniform(0.8, 1.2)
-            if (
-                not _is_retryable(resp)
-                or retry_num + 1 == _MAX_RETRYS
-                or backoff_seconds > (deadline_sec - time())
-            ):
-                _logger.error(
-                    "Failed to export metrics batch code: %s, reason: %s",
-                    resp.status_code,
-                    resp.text,
+
+        # If no batch size configured, export as single batch with retries as configured
+        if self._max_export_batch_size is None:
+            for retry_num in range(_MAX_RETRYS):
+                resp = self._export(
```

Comment:
What do you think about consolidating this block with the one on line 277? I think at least the retry logic can be factored out.

Reply:
Yes good idea! Extracted `_export_with_retries` in [d748c12](https://github.com/open-telemetry/opentelemetry-python/pull/4576/commits/d748c128cc8ea39f5875cfb0824286e4d5072f93)